pkg/operator/controller/route-status/route_status.go

```diff
@@ -0,0 +1,298 @@
+		return append(errs, fmt.Errorf("failed to list all routes in order to clear route status for deployment %s: %w", icName, err))
+	}
+	// Clear status on the routes that belonged to icName.
+	for i := range routeList.Items {
+		if cleared, err := ClearRouteStatus(kclient, &routeList.Items[i], icName); err != nil {
+			errs = append(errs, err)
+		} else if cleared {
+			routesCleared++
+		}
+	}
+	elapsed := time.Since(start)
+	log.Info("cleared all route status for ingress", "Ingress Controller",
+		icName, "Routes Status Cleared", routesCleared, "Time Elapsed", elapsed)
+
+	return errs
+}
+
+// ClearRouteStatus clears a route's status that is admitted by a specific ingress controller.
+func ClearRouteStatus(kclient client.Client, route *routev1.Route, icName string) (bool, error) {
+	if route == nil {
+		return false, fmt.Errorf("failed to clear route status: route is nil")
+	}
+	// Go through each route and clear status if admitted by this ingress controller.
+	var updated routev1.Route
+	for i := range route.Status.Ingress {
+		if condition := findCondition(&route.Status.Ingress[i], routev1.RouteAdmitted); condition != nil {
+			if route.Status.Ingress[i].RouterName == icName {
+				// Remove this status since it matches our routerName.
+				route.DeepCopyInto(&updated)
+				updated.Status.Ingress = append(route.Status.Ingress[:i], route.Status.Ingress[i+1:]...)
+				if err := kclient.Status().Update(context.TODO(), &updated); err != nil {
+					return false, fmt.Errorf("failed to clear route status of %s/%s for routerName %s: %w",
+						route.Namespace, route.Name, icName, err)
+				}
+				log.Info("cleared admitted status for route", "Route", route.Namespace+"/"+route.Name,
+					"Ingress Controller", icName)
+				return true, nil
+			}
+		}
+	}
+
+	return false, nil
+}
+
+// ClearRoutesNotAdmittedByIngress clears routes status that are not selected by a specific ingress controller.
+func ClearRoutesNotAdmittedByIngress(kclient client.Client, ingress *operatorv1.IngressController) []error {
+	var errs []error
+	if ingress == nil {
+		return append(errs, fmt.Errorf("ingress controller is nil"))
+	}
+
+	start := time.Now()
+
+	// List all routes.
+	routeList := &routev1.RouteList{}
+	if err := kclient.List(context.TODO(), routeList); err != nil {
+		return append(errs, fmt.Errorf("failed to list all routes in order to clear route status: %w", err))
+	}
+
+	// List all the Namespaces filtered by our ingress's Namespace selector.
+	namespacesInShard, err := GetNamespacesSelectedByIngressController(kclient, ingress)
+	if err != nil {
+		return append(errs, err)
+	}
+
+	// List routes filtered by our ingress's route selector.
+	routeSelector, err := metav1.LabelSelectorAsSelector(ingress.Spec.RouteSelector)
+	if err != nil {
+		return append(errs, fmt.Errorf("ingresscontroller %s has an invalid route selector: %w", ingress.Name, err))
+	}
+
+	// Iterate over the entire route list and clear if not selected by route selector OR namespace selector.
```

Comment:
```suggestion
	// Iterate over the entire route list and clear if either the route selector OR the namespace selector does not select it
```
This comment reads to me as it will clear the status if `!(routeInShard || namespaceInShard)`, which isn't consistent with the way it actually works. Maybe reword it to help prevent future confusion

Reply:
Done